### PR TITLE
 Remove alias gl calls from Batch and Introduce a ResetStage Batch command 

### PR DESCRIPTION
**Pull request by samcake**
- Remove all the glFunction aliases in Batch that we don;t need anymore
  Still need the glUniformXX calls to assign the uniform values
  Still need the LineWidth call, maybe could be part of the input stage explicitely ?
  Still need the Color call in MOdel.cpp waiting on core profile to rely on the attribute frequency support to remove it

- Most of the gl calls where used in the ResetGLState Job happening at the end of the pipeline. This is now replaced by a "resetStages" command on the Batch that implicitely reset everything from the gpu::Context, forget all tracked resource (Buffers, Textures, PIpeline, Framebuffer) and reset the actual glCOntext state to gpu::Context default

- REmoved the glEnbale calls about the Point Size feature and set it enable always

- Force writing mask on for depth when clearing depth buffer and then set it back to the cache value

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 66c5aec744273781b735c10b5d625d761d3b95b5 available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: baf7873ce13f9ba0279b89cdc9b0843640727922 available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by ZappoMan** [comment]
Mac build seems borked... 
screen shot 2015-07-28 at 9 40 40 am

**Comment by hifi-gustavo** [comment]
Build console output for commit: 22e70ee691c9126033d839692d812c984999eb1a available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 955b30ffd2ce19ae19529a303f85bf930c34bbaa available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: df9b66d2679c3740c8d885d5d02c2e37ea098e22 available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9cf3dfd3f7ef47469f87773daac5de87fba2b90f available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by jherico** [comment]
TEST THIS PLEASE

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9cf3dfd3f7ef47469f87773daac5de87fba2b90f available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8b7ad9cfcbaa68ab360c0a7561e5af5da8ee3c6e available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ac46b2bd1733ba18fe6e1db99a1ee204c7408bdc available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

File: libraries/gpu/src/gpu/GLBackend.cpp [hunk 1]
```diff
@@ -246,220 +227,49 @@ void GLBackend::do_drawIndexedInstanced(Batch& batch, uint32 paramOffset) {
...
-    DO_IT_NOW(_glBindBuffer, 2);
-}
-void GLBackend::do_glBindBuffer(Batch& batch, uint32 paramOffset) {
-    glBindBuffer(
-        batch._params[paramOffset + 1]._uint,
-        batch._params[paramOffset + 0]._uint);
-    (void) CHECK_GL_ERROR();
-}
+//#define DO_IT_NOW(call, offset) runLastCommand();
+#define DO_IT_NOW(call, offset)
```
Comment (ZappoMan) [inline libraries/gpu/src/gpu/GLBackend.cpp]: did you mean to do this?

**Comment by hifi-gustavo** [comment]
Build console output for commit: e32e45ed2b8faefc74c0f884d96eb9e1ee3a8865 available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 70d64a7777f90d6ae2afabc20348adfceb00e5e1 available at: https://gist.github.com/45ce36efdcc014e0915f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5439.dmg

**Pull request merged**